src/modules/auth/containers/FormFields.js

```diff
@@ -127,72 +135,140 @@ export function UserNameField(props) {
-						onBlur={regPwd1.onBlur}
-						innerRef={regPwd1.ref}
-					/>
-					{props.errors.password && <FormFeedback>{props.errors.password.message}</FormFeedback>}
-				</InputGroup>
-			</FormGroup>
+	// Password is watched for immediate feedback to the user
+	const watchedNewPassword = watch('newpassword', '');
+	const validateNewPassword = (value) => ({
+		minLength: validatePasswordLength(value, passwordCriteria?.minLength),
+		minLowercaseCount: validatePasswordLowercaseCount(value, passwordCriteria?.minLowercaseCount),
+		minUppercaseCount: validatePasswordUppercaseCount(value, passwordCriteria?.minUppercaseCount),
+		minDigitCount: validatePasswordDigitCount(value, passwordCriteria?.minDigitCount),
+		minSpecialCount: validatePasswordSpecialCount(value, passwordCriteria?.minSpecialCount),
+	});
 
-			<FormGroup>
-				<Label for="password2">{t("FormFields|Re-enter Password")}</Label>
-				<InputGroup>
-					<Input
-						id="password2"
-						name="password2"
-						type="password"
-						invalid={props.errors.password2}
-						autoComplete="new-password"
-						onChange={regPwd2.onChange}
-						onBlur={regPwd2.onBlur}
-						innerRef={regPwd2.ref}
-					/>
-					{props.errors.password2 && <FormFeedback>{props.errors.password2.message}</FormFeedback>}
-				</InputGroup>
-			</FormGroup>
-		</>
-	)
+	const regOldPassword = register('oldpassword');
+	const regNewPassword = register('newpassword', {
+		validate: {
+			passwordCriteria: (value) => (Object.values(validateNewPassword(value)).every(Boolean)
+			|| t('PasswordChangeField|Password does not meet security requirements')),
+			dontReuseOldPassword: (value) => (value !== getValues('oldpassword'))
+			|| t('PasswordChangeField|New password must be different from your old password'),
+		},
+	});
+	const regNewPasswordRepeat = register('newpassword2', {
+		validate: {
+			passEqual: value => (value === getValues().newpassword) || t('PasswordChangeField|Passwords do not match'),
+		},
+	});
+
+	return (<>
+		{currentPasswordInput && <FormGroup tag='fieldset' disabled={isSubmitting}>
+			<Label for='oldpassword'>
+				{t('PasswordChangeField|Current password')}{markRequired && '*'}
+			</Label>
+			<Input
+				autoFocus
+				id='oldpassword'
+				name='oldpassword'
+				type='password'
+				autoComplete='off'
+				required='required'
+				onChange={regOldPassword.onChange}
+				onBlur={regOldPassword.onBlur}
+				innerRef={regOldPassword.ref}
+			/>
+		</FormGroup>}
+		<FormGroup tag='fieldset' disabled={isSubmitting}>
+			<Label for='newpassword'>
+				{t('PasswordChangeField|New password')}{markRequired && '*'}
+			</Label>
+			<Input
+				id='newpassword'
+				name='newpassword'
+				type='password'
+				autoComplete='new-password'
+				required='required'
+				invalid={Boolean(errors?.newpassword)}
+				onBlur={regNewPassword.onBlur}
+				innerRef={regNewPassword.ref}
+				onChange={regNewPassword.onChange}
+			/>
+			{errors?.newpassword?.type !== 'passwordCriteria'
```

Comment:
@byewokko Ideally you should wrap the condition into parenthesis:

```
(errors?.newpassword?.type !== 'passwordCriteria')
```

Myabe we should add this into eslint criteria

Reply:
we dont have eslint active for this repository yet

i'll fix this one